File: easy-solution/galaxy4276/1346.java
import java.util.Arrays;

class Solution {

  /**
   * Input: arr = [10,2,5,3]
   * Output: true
   * Explanation: N = 10 is the double of M = 5,that is, 10 = 2 * 5.
   */
  public boolean checkIfExist(int[] arr) {
    boolean isAllZero = false;
    for (int i = 0; i < arr.length; i++) {
      if (arr[i] == 0) isAllZero = true;
      else isAllZero = false;
    }

    if (isAllZero) return true;

    for (int i = 0; i < arr.length; i++)
      for (int j = 0; j < arr.length; j++)
        if (arr[i] == arr[j] * 2 && (arr[i] != 0 && arr[j] != 0)) {
          return true;
        }
    return false;
  }
}

public class Application {
  public static void main(String[] args) {
    Solution s = new Solution();
    int[] nums = { -2,0,10,-19,4,6,-8 };
    boolean result = s.checkIfExist(nums);
    System.out.println("result: " + result);
  }
}

File: easy-solution/Antidote21/1346.c


bool checkIfExist(int* arr, int arrSize){
    int i,j,k=0;
    
    for(i=0;i<arrSize;i++){
        if(arr[i]==0){
            ++k;
            if(k==2){
                return true;
            }
        }
    }
    
    for(i=0;i<arrSize;i++){
        for(j=0;j<arrSize;j++){ //j=1로 변경해야 함 
            if(arr[i] == 2*arr[j]&&arr[i]!=0){ //arr[i] == 2*arr[j] ||  arr[j] == 2 * arr[i] 0이 두 개 일때도 성립 가능 
                return true;
            }
        }
        
    }
    
    return false;
}

bool checkIfExist(int* arr, int arrSize){
        for (int i = 0; i < arrSize; i++){
            for (int j = i+1; j < arrSize; j++){
                if(arr[i] == 2*arr[j] ||  arr[j] == 2 * arr[i]) return true; 
            }
        }
        return false;
    }
